_static/images/nuts_session-tokens.svg

```diff
@@ -0,0 +1,3 @@
erlimit="10" stroke-dasharray="3 3" pointer-events="stroke"/><path d="M 491.45 410.33 L 498.45 406.83 L 498.45 413.83 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><g transform="translate(513.5,398.5)"><switch><foreignObject style="overflow:visible;" pointer-events="all" width="70" height="11" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;font-size:11px;font-family:Helvetica;color:#000000;line-height:1.2;vertical-align:top;white-space:nowrap;text-align:left;"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;text-align:inherit;text-decoration:inherit;background-color:#ffffff;">check consent</div></div></foreignObject><text x="35" y="11" fill="#000000" text-anchor="middle" font-size="11px" font-family="Helvetica">check consent</text></switch></g><rect x="480" y="450" width="10" height="10" fill="#cc4125" stroke="#000000" pointer-events="all"/><path d="M 485.33 440.33 L 510.33 440.33 L 510.33 450.33 L 498.45 450.33" fill="none" stroke="#000000" stroke-miterlimit="10" stroke-dasharray="3 3" pointer-events="stroke"/><path d="M 491.45 450.33 L 498.45 446.83 L 498.45 453.83 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><g transform="translate(513.5,436.5)"><switch><foreignObject style="overflow:visible;" pointer-events="all" width="101" height="11" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;font-size:11px;font-family:Helvetica;color:#000000;line-height:1.2;vertical-align:top;white-space:nowrap;text-align:left;"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;text-align:inherit;text-decoration:inherit;background-color:#ffffff;">create and sign JWT</div></div></foreignObject><text x="51" y="11" fill="#000000" text-anchor="middle" font-size="11px" font-family="Helvetica">create and sign JWT</text></switch></g><ellipse cx="165" cy="47.5" rx="7.5" ry="7.5" fill="#cc4125" stroke="#000000" pointer-events="all"/><path d="M 165 55 L 165 80 M 165 60 L 150 60 M 165 60 L 180 60 M 165 80 L 150 100 M 165 80 L 180 100" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><g transform="translate(152.5,107.5)"><switch><foreignObject style="overflow:visible;" pointer-events="all" width="25" height="12" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;font-size:12px;font-family:Helvetica;color:#000000;line-height:1.2;vertical-align:top;white-space:nowrap;text-align:center;"><div xmlns="http://www.w3.org/1999/xhtml" style="display:inline-block;text-align:inherit;text-decoration:inherit;background-color:#ffffff;">User</div></div></foreignObject><text x="13" y="12" fill="#000000" text-anchor="middle" font-size="12px" font-family="Helvetica">User</text></switch></g></g></svg>
```

Comment:
Maybe split the diagram in 3 flows;
1. 'obtain session token' (first flow as depicted in diagram)
2. 'obtain access token' (second flow until FHIR GET call)
3. 'use access token/retrieve resources' (second from FHIR GET call to the end of the diagram).
That allows XIS to use the access token multiple times for the same subject.

Reply:
The validate call from XIS 2 to Node 2 is actually standardized as 'token introspection': https://tools.ietf.org/html/rfc7662
It's pretty simple to implement and broadly supported.